files/steps/generate_dataset_from_hbase.py

```diff
@@ -120,22 +121,32 @@ def encrypt_plaintext(data_key, plaintext_string, iv=None):
     return ciphertext.decode("ascii"), iv.decode("ascii")
 
 
+def get_key_from_cache(kek):
+    if kek in cache:
+        return cache[kek]
+
 def get_plaintext_key(url, kek, cek):
+    plaintext_key = None
+
     """Call DKS to return decrypted datakey."""
     request = retry_requests(methods=["POST"])
 
-    response = request.post(
-        url,
-        params={"keyId": kek, "correlationId": 0},
-        data=cek,
-        cert=(
-            "/etc/pki/tls/certs/private_key.crt",
-            "/etc/pki/tls/private/private_key.key",
-        ),
-        verify="/etc/pki/ca-trust/source/anchors/analytical_ca.pem",
-    )
-    content = response.json()
-    plaintext_key = content["plaintextDataKey"]
+    plaintext_key = get_key_from_cache(kek)
+
+    if not plaintext_key:
+        response = request.post(
+            DKS_DECRYPT_ENDPOINT,
```

Comment:
use `url` value as supplied to the get_plaintext_key() function

Reply:
done.